### PR TITLE
Adds property to global rules for header Image

### PR DESCRIPTION
**Pull request by yeah-sir**
This PR adds a global rule for header image

* Added a property for header image in GlobalRule
* Updated tests for GlobalRules

Follows #.
Related to #.
Fixes #.

### CODE REVIEW

File: src/Facebook/InstantArticles/Transformer/Rules/GlobalRule.php [hunk 1]
```diff
@@ -136,6 +139,21 @@ public function apply($transformer, $instantArticle, $node)
             $header->withTime(Time::create(Time::PUBLISHED)->withDatetime($timePublished));
         }
 
+        // Treats Header Image
+        $articleHeaderImageURL = $this->getProperty(self::PROPERTY_GLOBAL_HEADER_IMAGE, $node);
+        if ($articleHeaderImageURL) {
+            $header->withCover(Image::create()->withURL($articleHeaderImageURL));
+        } else {
```
Comment (diegoquinteiro) [inline src/Facebook/InstantArticles/Transformer/Rules/GlobalRule.php]: Maybe we don't need to add a warning here as the cover image is optional?

Comment (yeah-sir) [reply src/Facebook/InstantArticles/Transformer/Rules/GlobalRule.php]: Makes sense. I'll remove that block of code

File: src/Facebook/InstantArticles/Transformer/Rules/GlobalRule.php [hunk 2]
```diff
@@ -136,6 +139,12 @@ public function apply($transformer, $instantArticle, $node)
             $header->withTime(Time::create(Time::PUBLISHED)->withDatetime($timePublished));
         }
 
+        // Treats Header Image
+        $articleHeaderImageURL = $this->getProperty(self::PROPERTY_GLOBAL_HEADER_IMAGE, $node);
+        if ($articleHeaderImageURL) {
+            $header->withCover(Image::create()->withURL($articleHeaderImageURL));
```
Comment (everton-rosario) [inline src/Facebook/InstantArticles/Transformer/Rules/GlobalRule.php]: If this is the only way we are adding the image to the header, then, we are missing the opportunity to add caption content to this image as well.

Probably you will need to add a call to `$transformer->transform($header, $node)` or to find the caption by some other property you set on the configuration of the GlobalRule as well (some sort of `image.caption` property name to be created).

File: tests/Facebook/InstantArticles/Transformer/GlobalHtml/global.html [hunk 3]
```diff
@@ -14,6 +14,9 @@ <h1>The <b>article</b> title</h1>
                 <img src="http://domain.com/image-body.png"/>
                 <span class="caption">The caption for image.</span>
             </div>
+            <div class="hero-image">
```
Comment (everton-rosario) [inline tests/Facebook/InstantArticles/Transformer/GlobalHtml/global.html]: Consider adding a caption block of text to be added to the image on the header.
This way we will cover not only images, but also caption to these images.

File: src/Facebook/InstantArticles/Transformer/Rules/GlobalRule.php (hunk 2 above)
Comment (yeah-sir) [reply src/Facebook/InstantArticles/Transformer/Rules/GlobalRule.php]: Thanks @everton-rosario. I had spoken with @diegoquinteiro about this and he mentioned that image caption was low priority for now. @diegoquinteiro can you please confirm?

If we are considering image caption along with this change I will make sure make the appropriate code changes.

Comment (diegoquinteiro) [reply src/Facebook/InstantArticles/Transformer/Rules/GlobalRule.php]: @yeah-sir confirmed! Let's merge this one and follow up with an issue.

**Pull request merged**